### PR TITLE
Resolve most compiler warnings

### PR DESCRIPTION
**Pull request by serebit**
This is an in-progress PR that attempts to resolve most of the warnings thrown at compile time.

### CODE REVIEW

File: src/gs-window-x11.c [hunk 1]
```diff
@@ -2219,57 +2250,31 @@ create_panel (GSWindow *window)
...
-
-	fg.red = 0.8;
-	fg.green = 0.8;
-	fg.blue = 0.8;
-	fg.alpha = 1.0;
-
-	all_states = 0;
-
-	gtk_widget_override_background_color (window->priv->panel, all_states, &bg);
-	gtk_widget_override_color (window->priv->panel, all_states, &fg);
```
Comment (EbonJaeger) [inline src/gs-window-x11.c:2244]: Does this need to be replaced by anything, or is this just covered by GTK themes now?

File: src/gs-grab-x11.c [hunk 2]
```diff
@@ -317,7 +317,7 @@ gs_grab_move_mouse (GSGrab    *grab,
 
 	gs_debug ("*** releasing X server grab");
 	gdk_x11_ungrab_server ();
-	gdk_flush ();
+	gdk_display_flush (gdk_display_get_default ());
```
[the remark this answers is not in the document]
Comment (serebit) [reply src/gs-grab-x11.c:320]: `GdkDisplay` represents an X Display, which typically contains all attached monitors. As far as I can tell, this does the same thing.

File: src/gs-window-x11.c (hunk 1 above)
Comment (serebit) [reply src/gs-window-x11.c:2244]: Seems to be covered by GTK themes. I saw no visual difference when applying this patch.

File: src/gs-window-x11.c [hunk 3]
```diff
@@ -2219,57 +2250,31 @@ create_panel (GSWindow *window)
...
-	gtk_box_pack_start (GTK_BOX (window->priv->panel), alignment, FALSE, FALSE, 0);
 	window->priv->clock = gtk_label_new (NULL);
-	gtk_misc_set_padding (GTK_MISC (window->priv->clock), 4, 4);
-	gtk_container_add (GTK_CONTAINER (alignment), window->priv->clock);
-
-	right_alignment = gtk_alignment_new (1, 0.5, 1, 1);
-	gtk_box_pack_end (GTK_BOX (window->priv->panel), right_alignment, TRUE, TRUE, 0);
-	gtk_alignment_set_padding (GTK_ALIGNMENT (right_alignment),
-				   0, 0, 10, 10);
+	gtk_container_set_border_width (GTK_CONTAINER (window->priv->clock), 4);
```
[the remark this answers is not in the document]
Comment (serebit) [reply src/gs-window-x11.c]: Resolved, thanks.

**Review by EbonJaeger — APPROVED** [review]
Looks good to me. Everything seems to work for me as it did before (Intel/AMD combo).

**Pull request merged**